UefiTestingPkg/FunctionalSystemTests/MemoryProtectionTest/App/MemoryProtectionTestApp.c

```diff
@@ -897,59 +940,60 @@ UefiPageGuard (
     }
 
     UT_ASSERT_NOT_EFI_ERROR (mNonstopModeProtocol->ResetPageAttributes ());
+  } else if (MemoryProtectionContext.TestingMethod == MemoryProtectionTestReset) {
+    if (MemoryProtectionContext.TestProgress < 2) {
+      //
+      // Context.TestProgress indicates progress within this specific test.
+      // 0 - Just started.
+      // 1 - Completed head guard test.
+      // 2 - Completed tail guard test.
+      //
+      // We need to indicate we are working on the next part of the test and save our progress.
+      //
+      MemoryProtectionContext.TestProgress++;

```

Comment:
Just curious, why is the progress update move here? Is it expected that if the gBS->AllocatePages() call fails, MemoryProtectionContext.TestProgress would be 1?

Reply:
The structure of this looks the same as it did back in 2018. I think it works as written

Reply:
I see that now. The diff was not doing a great job lining things up. If the reset logic hasn't changed, then I won't look any further in this PR. I'll mark this resolved.